person.rb

```diff
@@ -0,0 +1,25 @@
+# person.rb
+
+class Person
+  attr_reader :id, :name, :age
+
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    @id = generate_id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age.to_i >= 18
+  end
+
+  def generate_id
+    "P#{rand(1000..9999)}"
+  end
+
+  # Add attr_writer for name and age
+  attr_writer :name, :age
```

Comment:
- [ ] Kindly move this to the top of the class for clarity and remove the `name` and` age `from the attribute reader. You can equally replace the `attribute writer `with `attribute accessor`